server.go

```diff
@@ -336,3 +342,17 @@ func (sb *SplitBytes) Next() bool {
 func (sb *SplitBytes) Chunk() []byte {
 	return sb.currentChunk
 }
+
+func (s *Server) registerPlugin(p plugin) {
+	s.pluginMtx.Lock()
+	s.plugins = append(s.plugins, p)
+	s.pluginMtx.Unlock()
```

Comment:
The `defer` seems a good idea!

Oh, I was curious about the mutex too. Seems like we mostly end up paying the cost of that mutex a bunch for very little gain. I didn't ask because it certainly seems _correct_ in that you definitely can't mutate `s.plugins` now. 😄